src/gui/generalsettings.cpp

```diff
@@ -139,6 +139,19 @@ GeneralSettings::GeneralSettings(QWidget *parent)
 {
     _ui->setupUi(this);
 
+    ConfigFile cfgFile;
+    _ui->callNotificationsCheckBox->setVisible(cfgFile.isComponentVisible());
```

Comment:
You can avoid calling the same method over and over, for instance:

```suggestion
    const ConfigFile cfgFile;
    const auto widgetVisible = cfgfile.isComponentVisible();
    _ui->callNotificationsCheckBox->setVisible(widgetVisible);
```

You can then use widgetVisible for all setVisible calls